src/Layout/Presentation/SingleDistrict/SingleDistrict.tsx

```diff
@@ -39,14 +41,19 @@ export class SingleDistrict extends React.Component<SingleDistrictProps, {}> {
     };
 
     render() {
-        const currentDistrictResult = this.getDistrictResult(this.props.districtSelected);
+        const currentDistrictResult = this.getDistrictResult(this.props.districtSelected)!;
+        const data = this.getData()!;
+        const partyResultMap = createPartyResultMap(data);
         const calculateVulnerable =
             isQuotientAlgorithm(this.props.algorithm) && currentDistrictResult.districtSeats > 0;
-        const vulnerableMap = calculateVulnerable ? getVotesToVulnerableSeatMap(currentDistrictResult!) : undefined;
-        const quotientMap = calculateVulnerable ? getQuotientsToVulnerableSeatMap(currentDistrictResult!) : undefined;
-        const vulnerable = calculateVulnerable ? getVulnerableSeatByQuotient(currentDistrictResult!) : undefined;
-        const vulnerableVotes = calculateVulnerable ? getVulnerableSeatByVotes(currentDistrictResult!) : undefined;
-        const data = this.getData()!;
+        const vulnerableMap = calculateVulnerable ? getVotesToVulnerableSeatMap(currentDistrictResult) : undefined;
+        const quotientMap = calculateVulnerable ? getQuotientsToVulnerableSeatMap(currentDistrictResult) : undefined;
+        const vulnerable = calculateVulnerable
+            ? getVulnerableSeatByQuotient(currentDistrictResult, partyResultMap, this.props.districtThreshold)
+            : undefined;
```

Comment:
Codacy found an issue: [Unexpected use of undefined.](https://app.codacy.com/gh/Project-Lavinia/Lavinia-client/pullRequest?prid=5991602)